ColorzCore/Parser/Macros/IsDefined.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using ColorzCore.Lexer;
+
+namespace ColorzCore.Parser.Macros
+{
+    class IsDefined : BuiltInMacro
+    {
+        public EAParser ParentParser { get; private set; }
+
+        public IsDefined(EAParser parent)
+        {
+            ParentParser = parent;
+        }
+
+        public override IEnumerable<Token> ApplyMacro(Token head, IList<IList<Token>> parameters)
+        {
+            if (parameters[0].Count != 1)
+            {
+                // TODO: err somehow
+                yield return MakeFalseToken(head.Location);
+            }
+            else
+            {
+                Token token = parameters[0][0];
+
+                if ((token.Type == TokenType.IDENTIFIER) && IsReallyDefined(token.Content))
+                {
+                    yield return MakeTrueToken(head.Location);
+                }
+                else
+                {
+                    yield return MakeFalseToken(head.Location);
+                }
+            }
+        }
+
+        public override bool ValidNumParams(int num)
+        {
+            return num == 1;
+        }
+
+        protected bool IsReallyDefined(string name)
+        {
+            return ParentParser.Definitions.ContainsKey(name) || ParentParser.Macros.ContainsName(name);
```

Comment:
Considering #ifdef does this, it seems like checking for label/variable definition would be moer useful. Just wanna hear your thoughts.

Reply:
I'm just following the general behaviour of the nl!Core impl, which doesn't (and couldn't even if it wanted to, because of the nature of its preprocessor) care for labels.

(Plus I don't think I like the idea of mixing preprocessor (macros) stuff and core language stuff (labels). If we want something like this for labels maybe it could be implemented as some operator instead?)